src/components/esphome-header-menu.ts

```diff
@@ -74,21 +75,28 @@ export class ESPHomeHeaderMenu extends LitElement {
     this._isWide = isWideListener.matches;
   };
 
-  private _handleUpdateAll() {
-    if (!confirm(`Do you want to update all devices?`)) {
+  private async _handleUpdateAll() {
+    if (
+      !(await showConfirmationDialog({
+        title: "Update All",
+        text: "Do you want to update all devices?",
+        confirmText: "Update All",
+        dismissText: "Cancel",
+      }))
+    ) {
       return;
     }
-    openUpdateAllDialog();
+    openUpdateAllProcessDialog();
   }
 
   private _handleEditSecrets() {
     openEditDialog(SECRETS_FILE);
   }
 
-  private _handleOverflowAction(ev: CustomEvent<ActionDetail>) {
+  private async _handleOverflowAction(ev: CustomEvent<ActionDetail>) {
     switch (ev.detail.index) {
       case 0:
-        this._handleUpdateAll();
+        await this._handleUpdateAll();
```

Comment:
VS Code suggested this, not sure if _handleEditSecrets should be async too

Reply:
awaiting it only helps with finding the stack trace when things blow up, but it doesn't really matter here. We don't care about the result.

Reply:
I've removed the await keyword.